src/lib/bootloader/autoyast_converter.rb

```diff
@@ -149,16 +149,9 @@ def import_stage1(data, bootloader)
       end
 
       def import_stage1_devices(data, stage1)
-        # extended partion maybe do not exists, so report it to user
-        if (data["global"]["boot_extended"] == "true" ||
-            data["location"] == "extended") &&
-            Yast::BootStorage.extended_partition.nil?
-          raise "boot_extended used in autoyast profile, but there is no extended partition"
-        end
-
-        STAGE1_DEVICES_MAPPING.each do |key, device|
+        STAGE1_DEVICES_MAPPING.each do |key, method|
           if data["global"][key] == "true" || data["boot_#{key}"]
-            stage1.add_udev_device(Yast::BootStorage.public_send(device).name)
+            stage1.add_udev_device(stage1.public_send(method))
```

Comment:
This new line is not covered by tests.

Can this PR break previously valid AY profiles?

Reply:
no. In fact as you can see in module we no longer export this value ( we do not export it even in SLE12 ). It is used mainly as workaround when bootloader propose wrong value and user want explicitly overwrite it.

But I can try to add test for this code.

Reply:
test added and code fixed